app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -158,6 +164,83 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
                 throw new RuntimeException(e);
             }
         });
+
+        binding.requestNewShare.setOnClickListener(view1 -> {
+            requireActivity().runOnUiThread(() -> {
+                String userAgent = new WebView(getContext()).getSettings().getUserAgentString();
+                try {
+                    SharetransferModule.requestNewShare(activity.appKey, userAgent, activity.appKey.getShareIndexes().get(0), (result) -> {
+                        if(result instanceof Result.Error) {
```

Comment:
@metalurgical I am not very sure about the third argument (`activity.appKey.getShareIndexes().get(0)`) that I passed to requestNewShare. It works for me though. Method signature suggests to pass `String availableShareIndexes` which is plural. Is that correct ?

Reply:
It can be an empty array "[]".

Can you share how you tested this for thresholdkey instance on device A and thresholdkey instance on device B?